src/jit/decomposelongs.cpp

```diff
@@ -1112,6 +1112,24 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
         {
             case GT_LSH:
             {
+                // Reduce count modulo 64 to match behavior found in
+                // the LLSH helper and in gtFoldExpr.
+                count &= 0x3F;
```

Comment:
Shouldn't this fix be enabled only on x86? AFAIR arm32 has a different behavior.

Reply:
`gtFoldExpr` doesn't do anything different for arm32 / x86, it always reduces. Also the left shift helper for arm uses C++ unsigned left shift semantics, which also reduce.

So I think we're ok with this change on arm32 too, but I'll investigate further.

Reply:
> Also the left shift helper for arm uses C++ unsigned left shift semantics, which also reduce.

The behavior of C++ shift operations when the shift count exceeds the bid width is undefined.

Reply:
The arm register LSL instruction uses the low byte of its register count argument as the shift count; slightly different from xarch.

Reply:
> gtFoldExpr doesn't do anything different for arm32 / x86, it always reduces.

Ah, forgot about that little detail. `ValueNumStore::EvalOpIntegral` never reduces so you get whatever the C++ compiler thinks it's cool.

Reply:
Seems like we should also update the C++ LLSH helper to mask with 0x3F to be consistent  with non-helper usage and avoid the potential of relying on compiler-defined behavior.